app/email.go

```diff
@@ -521,14 +521,19 @@ func (es *EmailService) SendDeactivateAccountEmail(email string, locale, siteURL
 }
 
 func (es *EmailService) SendRemoveExpiredLicenseEmail(email string, locale, siteURL string, licenseId string) *model.AppError {
+	renewalToken, e := es.srv.GenerateRenewalToken(JWTDefaultTokenExpiration)
+	if e != nil {
+		return e
+	}
+
 	T := utils.GetUserTranslations(locale)
 	subject := T("api.templates.remove_expired_license.subject",
 		map[string]interface{}{"SiteName": es.srv.Config().TeamSettings.SiteName})
 
 	bodyPage := es.newEmailTemplate("remove_expired_license", locale)
 	bodyPage.Props["SiteURL"] = siteURL
 	bodyPage.Props["Title"] = T("api.templates.remove_expired_license.body.title")
-	bodyPage.Props["Link"] = fmt.Sprintf("%s?id=%s", model.LICENSE_RENEWAL_LINK, licenseId)
+	bodyPage.Props["Link"] = fmt.Sprintf("%s?token=%s&id=%s", LicenseRenewalURL, renewalToken, licenseId)
```

Comment:
No need to send the license Id because it goes inside the token

Can you also point me to the LicenseRenewalURL variable assignment?

Reply:
@ethervoid https://github.com/mattermost/mattermost-server/blob/3da6f270ec356b93551059b9ff6af43db93b590a/app/license.go#L24